File: Sorting/Selection_Sort/selection_sort.c
#include<stdio.h>
int main()
{
    int n,i=0,j;
    printf("Enter the length of List : ");
    scanf("%d",&n);
    int arr[n];
    printf("Enter the numbers of list : \n");
    while(i<n)
    {
        scanf("%d",&arr[i]);
        i++;
    }
    selectionSort(arr,n);
    return 0;
}
void selectionSort(int list[],int n)
{
    for(int i=0;i<n-1;i++)
    {
        int min_index=i;
        for(int j=i+1;j<n;j++)
        {
            if(list[j]>list[min_index])
            {
                min_index=j;
            }
        }
        int temp=list[i];
        list[i]=list[min_index];
        list[min_index]=temp;
    }
    printf("Sorted Array : ");
    for(int i=0;i<n;i++)
    {
        printf(" %d ",list[i]);
    }
}

File: Sorting/Selection_Sort/selection_sort.py
def selectionSort(n,list):
	for i in range(n-1):
		min_index=i
		for j in range(i+1,n):
			if list[j]<list[min_index]:
				min_index=j
		#swapping
		temp=list[i]
		list[i]=list[min_index]
		list[min_index]=temp
	print("Sorted array : ",end=' ')
	print(list)

def main():
	list=[]
	print("Enter the numbers of the list (Enter 000 to stop) : ")
	while True:
		num=int(input())
		if num==000:
			break
		list.append(num)
	selectionSort(len(list),list)

if __name__ == '__main__':
	main()
